Use common utility functions over duplicate implementation in blockchain.rs

mitch-fuel opened the pull request:
A function was created in blockchain.rs to reduce duplication validation logic that was then extracted in subsequent PRs. But because the blockchain.rs testing PR wasn't merged in yet, this refactor PR became necessary.

xgreenx reviewed (dismissed):
Why the history of commits is so strange?=D Maybe you can transform into one commit?

mitch-fuel commented:
> Why the history of commits is so strange?=D Maybe you can transform into one commit?

fixed, this issue started off based on a branch and depended on other branches coming in first. I rebased it into 1 commit
